docs-search/client-halogen/spago.yaml

```diff
@@ -0,0 +1,9 @@
+package:
+  name: docs-search-client-halogen
+  dependencies:
+    - docs-search-common
+    - halogen
+    - halogen-css
+    - js-uri
+    - markdown-it-halogen
+    - profunctor-lenses
```

Comment:
I believe old spago was more picky about listing out dependencies, than new spago is.
Basically, what I did to construct this list:

1. Copy over all the deps from `docs-search/spago.yaml`
2. Clear built artifacts: `rm -rf .spago output`
3. Comment out all the deps here
4. Attempt to build: `spago build -p docs-search-client-halogen`
5. Renable a package to fix a build error, then go to step 4.

And surprisingly the package builds even though required packages like `prelude` are not listed here.

Old spago would warn if an imported module was not listed as a package dependency. In new spago, I'm able to omit deps which are pulled in transitively. And that doesn't seem intended.

Reply:
The behaviour is still there, it's just better in the new Spago:
- it's optional rather than mandatory, and you'll need the `pedantic-packages` flag/config bit to make it check that your packages are exactly the ones you need
- and when it complains, it gives you pointers to modules
- when publishing a package this is however turned on by default, so we only publish "correct" dependencies

Reply:
I would be ok with enabling `pedantic-packages` in the global config, so we always detect incorrect dependencies

Reply:
Ok, thanks for the heads up. The UX is indeed improved!
I liked being able to copy the commands directly from the terminal. 

I noticed that its pedantic about test deps too.
I'm straying from the original topic, but can I suggest a change in behavior regarding test deps?

I want spago to treat the test modules deps as if they were declared as the union of the project deps and the test deps.
 
That is, it would still pedantically warn if I screw up the new test dependency list, but it wouldn't require me to reiterate project deps.

Reply:
I noticed that its pedantic about test deps too.
I'm straying from the original topic, but can I suggest a change in behavior regarding test deps?

I want spago to treat the test modules deps as if they were declared as the union of the project deps and the test deps.
 
That is, it would still pedantically warn if I screw up the new test dependency list, but it wouldn't require me to reiterate project deps.

Reply:
It's possible, I may have jumped to conclusions too rapidly. 
When running `spago build --pedantic-package -p spago`.

I got output this output: 
```console
  Run the following command to install them all:
  spago install -p spago aff-promise ansi console control enums exceptions formatters node-fs-aff routing-duplex unfoldable unicode
  
  ...
  
  Run the following command to install them all:
  spago install --test-deps -p spago console enums identity newtype node-fs-aff quickcheck

```

So all that is really happening is that the missing deps are reported for both the project and test deps.

Reply:
I may still open an issue for this. I have two semi related problems:
(1) Re-reporting deps that belong in the project dep list and not the test deps will lead to users doing the wrong thing.
(2) I would like to see all the recommended install commands output at the bottom of the terminal output. Scrolling up and down the list to copy the commands is a bit painful.

Reply:
@CharlesTaylor7 ah I see now - agreed that these should both be fixed 👍